docs/proposals/secrets-proposal.md

```diff
@@ -0,0 +1,133 @@
+## Proposal for supporting secrets in OpenCompose
+
+### Kubernetes Secrets
+
+A very simple secret object in Kubernetes looks like this -
+
+```yaml
+apiVersion: v1
+kind: Secret
+metadata:
+  name: kubesec
+type: Opaque
+data:
+  username: <base64 encoded value>
+  password: <base64 encoded value>
+```
+
+Note that, as seen above, one secret can have multiple `data` values.
+
+---
+
+#### Creating a new secret:
+
+To create a new secret, a top level `secrets` key can be used as follows:
+
+ ```yaml
+version: "0.1-dev"
+
+services:
+   name: foobar
+   containers:
+   - image: foo/bar:tag
+     ports:
+     - port: 8080:80
+     env:
+     - foo=bar
+     - travis_creds:
+         secret: ci_secret/travis
+     mounts:
+     - mountPath: /etc/jenkins.conf
+       secret: ci_secret/jenkins
+
+secrets:
+    ci_secret:
+      travis:
+        file: /etc/travis.passwd
+      jenkins:
+        data: strongpassword@@
+```
+
+The syntax will look like -
+
+```yaml
+secrets:
+  secret_name:
+    secret_key:
+      data: secret_value
+```
+
+In order to retrieve a secret from a file, use -
+
+```yaml
+secrets:
+  secret_name:
+    secret_key:
+      file: <file path>
+```
+
+---
+
+
+To consume the above secret in OpenCompose, here is the proposed syntax.
+
+Secrets in Kubernetes can be consumed either as an environment variable, or as a mounted volume.
+ 
+- ##### Consuming an already created Kubernetes secret in OpenCompose as an environment variable
+ 
+ ```yaml
+version: "0.1-dev"
+
+services:
+   name: foobar
+   containers:
+   - image: foo/bar:tag
+     ports:
+     - port: 8080:80
+     env:
+     - foo=bar
+     - db_user:
+         secret: kubesec/username
+     - db_pass:
+         secret: kubesec/password
+```
+
+- ##### Consuming an already created Kubernetes secret in OpenCompose as a mounted volume
+ 
+ ```yaml
+version: "0.1-dev"
+
+services:
+   name: foobar
+   containers:
+   - image: foo/bar:tag
+     ports:
+     - port: 8080:80
+     mounts:
+     - mountPath: /var/secret_path/user_info
+       secret: kubesec/username
```

Comment:
`secretName` might be better as it will be consistent with how we reference Volumes here (`volumeName`).